publishing-sdk/src/main/java/com/ably/tracking/publisher/guards/DublicateTrackableGuard.kt

```diff
@@ -50,7 +50,7 @@ internal class DefaultDuplicateTrackableGuard : DuplicateTrackableGuard {
      */
     override fun finishAddingTrackable(trackable: Trackable, result: Result<AddTrackableResult>) {
         trackablesCurrentlyBeingAdded.remove(trackable)
-        duplicateAddCallsHandlers[trackable]?.forEach { handler -> handler(result) }
+        duplicateAddCallsHandlers[trackable]?.iterator()?.forEach { handler -> handler(result) }
```

Comment:
@AndyTWF I wonder whether here, and for the other instance, it would be appropriate to fail early and NPE if `iterator()` returns `null` - that is, `iterator()!!` rather than `iterator()?`. 🤔

Reply:
Done in https://github.com/ably/ably-asset-tracking-android/pull/915/commits/b64f88ff541923af05edab847e842dc95488ce52

Reply:
@QuintinWillison I had to revert this as the !! operator appears to check the entire result of the expression (so if we get a NULL from `duplicateAddCallsHandlers[trackable]`, it still throws the NPE)

Reply:
Thanks! 😁 FYI, @KacperKluka and @davyskiba, in case you have any further thoughts on this.

Reply:
If we want to have the NPE we could use `?.let {}` to get a non-null list of handlers and then use `iterator()!!` to fail early in case of a NPE 😉 It would look like this:
```suggestion
        duplicateAddCallsHandlers[trackable]?.let { it.iterator()!!.forEach { handler -> handler(result) } }
```
I'd also add a comment that this is intentional because we want to fail early on NPEs as otherwise someone might be tempted to change it to null-safe calls in the future 😅

Reply:
Thanks, @KacperKluka. That looks like the right balance, and good point regarding commentary. I'll push a commit shortly for this.

Reply:
Once the `let` was inserted it became clear that `iterator()` itself, in isolation, can't return `null` (oops, should have checked that API!). No need for explicit NPE injection after all. 😁 https://github.com/ably/ably-asset-tracking-android/pull/915/commits/d5f6719737c53fef046da6eb0d8f3310cd7b90b0